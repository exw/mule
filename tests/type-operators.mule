let type t a = int in
(4 : t {})
