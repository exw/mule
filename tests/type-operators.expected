
Desugared:

	(LetType
	 (((t (TypeLam (() a (Recur (() t (Var (() int))))))))
	  (App
	   ((WithType
	     (App
	      (() (Var (() t))
	       (Record (r_info ()) (r_types (() () ())) (r_values (() () ()))))))
	    (Const (Integer 4))))))

inferred type:

	(Named (() int))

Runtime term:

	(Integer 4)

Evaluated:

	(Integer 4)
